src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -104,8 +112,11 @@ def update_vars_and_postname(self):
 
         '''
         suffix = ""
-        while any(out_var+self._post_name+str(suffix) in self._input_list
-                  for out_var in self._output_list):
+        # The signatures in the input/output list need to be converted
+        # back to strings to easily append the suffix
+        input_string = [str(input_var) for input_var in self._input_list]
+        while any(str(out_sig)+self._post_name+str(suffix) in input_string
```

Comment:
I've added a comment (I also rewrote the docstring above a bit to make it clearer, forgot to mention that). This code will actually work for derived type (e.g. if a user input variable `a%b_post` exists, and we would have an output variable `a%b`, this would create a clash in the NetCDF file (both creating `a%b_post` as a variable name in the NetCDF file).